### PR TITLE
fix: add management command to update incorrect opportunity id

### PR DESCRIPTION
**Pull request by muhammad-ammar**
**JIRA:** https://openedx.atlassian.net/browse/ENT-4701

**Description:** Update incorrect opportunity id for university of iceland manual enrollment order enterprise offer.

## Description

Describe what this pull request changes, and why these changes were made. How will these changes affect other people, installations of edx, etc.?
Please include links to any relevant ADRs, design artifacts, and decision documents. Make sure to document the rationale behind significant changes in the repo, per [OEP-19](https://open-edx-proposals.readthedocs.io/en/latest/oep-0019-bp-developer-documentation.html), and can be
linked here.

Useful information to include:
- Which edX user roles will this change impact? Common user roles are "Learner", "Course Author", "Developer", and "Operator".
- Include screenshots for changes to the UI (ideally, both "before" and "after" screenshots, if applicable).
- Provide links to the description of corresponding configuration changes. Remember to correctly annotate these changes.

## Supporting information

Link to other information about the change, such as Jira issues, GitHub issues, or Discourse discussions.
Be sure to check they are publicly readable, or if not, repeat the information here.

## Testing instructions

Please provide detailed step-by-step instructions for testing this change; how did YOU test this change?

## Other information

Include anything else that will help reviewers and consumers understand the change.
- Does this change depend on other changes elsewhere?
- Any special concerns or limitations? For example: deprecations, migrations, OpenEdx vs. edx.org differences, development vs. production environment differences, security, or accessibility.

### CODE REVIEW

**Review by saleem-latif — APPROVED** [review]

File: ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            type=str,
+        )
+
+    def handle(self, *args, **options):
+        logger.info('Started update of opportunity id.')
+
+        with open(options['data_csv']) as csv_file:
+            reader = csv.DictReader(csv_file)
+            for row in reader:
+                manual_order_enterprise_offer = ConditionalOffer.objects.get(
```
Comment (HammadAhmadWaqas) [inline ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py:45]: Can we add error handling here? For example what if there are no conditional offer found or there are multiple conditional offer found

File: ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+ConditionalOffer = get_model('offer', 'ConditionalOffer')
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class Command(BaseCommand):
+    """
+    Update opportunity id.
```
Comment (HammadAhmadWaqas) [inline ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py]: Can we add more details about the command? Like what it does and how. We could add an example command to run it.

File: ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            dest='data_csv',
+            default=None,
+            help='Path of csv to read enterprise uuids and opportunity ids.',
+            type=str,
+        )
+
+    def handle(self, *args, **options):
+        logger.info('Started update of opportunity id.')
+
+        with open(options['data_csv']) as csv_file:
```
Comment (HammadAhmadWaqas) [inline ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py:42]: `data_csv` is not a required param. I think command will break if user will not provide this param.

File: ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+class Command(BaseCommand):
+    """
+    Update opportunity id.
+    """
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--data-csv',
+            action='store',
+            dest='data_csv',
+            default=None,
```
Comment (HammadAhmadWaqas) [inline ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py]: Why we are not making this param as required? Will this command work without this param?

File: ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        with open(options['data_csv']) as csv_file:
+            reader = csv.DictReader(csv_file)
+            for row in reader:
+                manual_order_enterprise_offer = ConditionalOffer.objects.get(
+                    offer_type=ConditionalOffer.USER,
+                    priority=OFFER_PRIORITY_MANUAL_ORDER,
+                    condition__enterprise_customer_uuid=UUID(row['enterprise_uuid']),
+                )
+                logger.info(
+                    'Offer Name: [%s], Old Opportunity ID: [%s], New Opportunity ID: [%s]',
```
Comment (HammadAhmadWaqas) [inline ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py]: Can we add prefixes to this logger so we can track them easily?

File: ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+                    manual_order_enterprise_offer.name,
+                    manual_order_enterprise_offer.sales_force_id,
+                    row['opportunity_id']
+                )
+                manual_order_enterprise_offer.sales_force_id = row['opportunity_id']
+                manual_order_enterprise_offer.save()
+
+        logger.info('Finished update of opportunity id.')
```
Comment (HammadAhmadWaqas) [inline ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py]: Can we log the final statistics at the end of the command to show how many records we updated and how many failures occurred?

File: ecommerce/enterprise/tests/test_update_enterprise_offer_opportunity_id.py [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        call_command(
+            'update_enterprise_offer_opportunity_id', '--data-csv={}'.format(csv_file_path)
+        )
+
+        offer = ConditionalOffer.objects.get(
+            offer_type=ConditionalOffer.USER,
+            priority=OFFER_PRIORITY_MANUAL_ORDER,
+            condition__enterprise_customer_uuid=self.enterprise_customer
+        )
+        self.assertEqual(offer.sales_force_id, self.correct_opportunity_id)
```
Comment (HammadAhmadWaqas) [inline ecommerce/enterprise/tests/test_update_enterprise_offer_opportunity_id.py:72]: can we confirm the incorrect opportunity before the command run

**Review by HammadAhmadWaqas — CHANGES_REQUESTED** [review]

File: ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py (hunk 1 above)
Comment (muhammad-ammar) [reply ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py:45]: there is only one manual enrollment order offer that we need to update and it already exists

File: ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py (hunk 6 above)
Comment (muhammad-ammar) [reply ecommerce/enterprise/management/commands/update_enterprise_offer_opportunity_id.py]: CSV file will have opportunity id for single enterprise offer

**Review by HammadAhmadWaqas — APPROVED** [review]

**Review by zamanafzal — APPROVED** [review]

**Pull request merged**